.github/workflows/validate_changed_files.sh

```diff
@@ -7,10 +7,14 @@ VALID_FILE_PATHS=('pkg/plugin/sdk/v1alpha2') # Array of filepaths that are allow
 
 # by default the only local branch will be pull/PR#/merge
 # fetch only the latest commit from the 2 branches in question to avoid fetching the entire repo which could be costly
-git fetch --depth 1 origin "$1"
-git fetch --depth 1 origin "$2"
 
-git diff "origin/$1..origin/$2" --name-only | while read -r file; do
+git fetch --depth 1 origin $GITHUB_BASE_REF
+git fetch --depth 1 origin $GITHUB_REF
+
+main_head=`git rev-parse --short origin/$GITHUB_BASE_REF`
+pr_head=$GITHUB_SHA
+
+git diff --name-only ${main_head}..${pr_head} | while read -r file; do
```

Comment:
In this case, it doesn't matter, but to make a habit, always wrap Bash variables with double quotes to avoid word splitting.
```suggestion
git fetch --depth 1 origin "${GITHUB_BASE_REF}"
git fetch --depth 1 origin "${GITHUB_REF}"

main_head=$(git rev-parse --short "origin/${GITHUB_BASE_REF}")
pr_head="${GITHUB_SHA}"

git diff --name-only "${main_head}..${pr_head}" | while read -r file; do
```